### ISSUE TITLE
model.train should keep history?

### ISSUE THREAD

**Issue by lacan** [CSBDeep/CSBDeep#38]
Hello, 

I know that if we want to plot or have a look at the history of a training, we can do 
` history = model.train( ... )`

But would it not be more interesting for the Model class to store the history object internally? Then we could access it via `history = model.getHistory()`
If the history could be serialized to JSON as well, then we could always load a previous model and see the stats of how the training went?
Is that something that is already there? If not I'd be happy to discuss its use here, and eventually work out how to integrate it into CSBDeep / StarDIST.

**Comment by uschmidt83** [comment]
> If the history could be serialized to JSON as well, then we could always load a previous model and see the stats of how the training went?

That seems the best option to me. I assume you want to save the loss and metrics history, which is stored in `history.history`?

> Is that something that is already there?

That's not part of the current library, since we typically use TensorBoard to look at the history and don't really use the Keras `history` much.

> If not I'd be happy to discuss its use here, and eventually work out how to integrate it into CSBDeep / StarDIST.

I found this callback that I had written some time ago, which serializes the history after every epoch during training:

```python
import os, json
import numpy as np
import keras.backend as K
from keras.callbacks import Callback

class HistoryCheckpoint(Callback):
    def __init__(self, modelpath, fname='history.json', save_lr=True):
        assert os.path.exists(modelpath)
        self.filepath = os.path.join(modelpath,fname)
        self.save_lr = save_lr

    def on_train_begin(self, logs=None):
         self.history = {}

    def on_epoch_end(self, epoch, logs=None):
        logs = logs or {}
        if self.save_lr:
            logs['lr'] = K.get_value(self.model.optimizer.lr)
        for k,v in logs.items():
            self.history.setdefault(k,[]).append(np.float64(v))
        if self.history:
            with open(self.filepath,'w') as f:
                f.write(json.dumps(self.history))
```

Would this be good enough?

Best,
Uwe
